pkg/model/firewall.go

```diff
@@ -141,6 +141,13 @@ func (b *FirewallModelBuilder) applyNodeToMasterAllowSpecificPorts(c *fi.ModelBu
 			tcpPorts = append(tcpPorts, 179)
 			protocols = append(protocols, ProtocolIPIP)
 		}
+
+		if b.Cluster.Spec.Networking.Romana != nil {
+			// Romana needs to access etcd
+			glog.Warningf("Opening etcd port on masters for access from the nodes, for romana.  This is unsafe in untrusted environments.")
+			tcpPorts = append(tcpPorts, 4001)
```

Comment:
It looks like you open etcd on 12379, not 4001.

Reply:
Kind of. The service port is 12379, but that gets translated by kube-proxy's iptables rules to a specific IP and port 4001. So the security group sees connections to port 4001.